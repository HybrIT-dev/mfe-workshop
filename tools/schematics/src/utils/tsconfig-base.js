"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateTsBaseConfig = void 0;
const devkit_1 = require("@nrwl/devkit");
function updateTsBaseConfig(tree, projectName, projectRoot) {
    const tsConfig = 'tsconfig.base.json';
    try {
        (0, devkit_1.updateJson)(tree, tsConfig, (tsConfig) => {
            Object.assign(tsConfig.compilerOptions.paths, {
                [`@senses/${projectName}`]: [
                    (0, devkit_1.joinPathFragments)(projectRoot, 'index.ts'),
                ],
            });
            return tsConfig;
        });
    }
    catch (error) {
        throw new Error(`Unable to update ${tsConfig}`);
    }
}
exports.updateTsBaseConfig = updateTsBaseConfig;
//# sourceMappingURL=tsconfig-base.js.map