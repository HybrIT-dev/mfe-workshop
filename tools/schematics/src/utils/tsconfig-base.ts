import { Tree, joinPathFragments, updateJson } from '@nrwl/devkit';
import { CompilerOptions } from 'typescript';

export function updateTsBaseConfig(
  tree: Tree,
  projectName: string,
  projectRoot: string
) {
  const tsConfig = 'tsconfig.base.json';
  try {
    updateJson(
      tree,
      tsConfig,
      (tsConfig: Record<'compilerOptions', CompilerOptions>) => {
        Object.assign(tsConfig.compilerOptions.paths, {
          [`@senses/${projectName}`]: [
            joinPathFragments(projectRoot, 'index.ts'),
          ],
        });
        return tsConfig;
      }
    );
  } catch (error) {
    throw new Error(`Unable to update ${tsConfig}`);
  }
}
