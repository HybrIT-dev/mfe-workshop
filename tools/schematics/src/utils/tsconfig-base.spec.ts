import { Tree } from '@nrwl/devkit';
import { createTree } from '@nrwl/devkit/testing';
import { updateTsBaseConfig } from './tsconfig-base';

describe('tsconfig-base utils', () => {
  let workspaceTree: Tree;

  beforeEach(async () => {
    workspaceTree = createTree();
    workspaceTree.write(
      'tsconfig.base.json',
      JSON.stringify({ compilerOptions: { paths: {} } })
    );
  });

  describe('updateTsBaseConfig', () => {
    it('should throw an error when there is an exception during the update', () => {
      workspaceTree.delete('tsconfig.base.json');

      expect(() =>
        updateTsBaseConfig(workspaceTree, 'my-new-app', 'apps/my-new-app/src')
      ).toThrowError('Unable to update tsconfig.base.json');
    });

    it('should create an entry for the project', () => {
      updateTsBaseConfig(workspaceTree, 'my-new-app', 'apps/my-new-app/src');

      const tsconfigBaseJson = workspaceTree.read(
        'tsconfig.base.json',
        'utf-8'
      );
      expect(tsconfigBaseJson).toMatchSnapshot();
    });
  });
});
