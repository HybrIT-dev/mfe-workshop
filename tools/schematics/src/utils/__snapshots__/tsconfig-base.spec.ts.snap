// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsconfig-base utils updateTsBaseConfig should create an entry for the project 1`] = `
"{
  \\"compilerOptions\\": {
    \\"paths\\": {
      \\"@senses/my-new-app\\": [
        \\"apps/my-new-app/src/index.ts\\"
      ]
    }
  }
}
"
`;
